napari/components/add_layers_mixin.py

```diff
@@ -896,3 +900,44 @@ def _add_layer_from_data(
                 raise exc
 
         return layer
+
+    def _save_individual_layers_with_plugin(
```

Comment:
@tlambert03 one problem here is that the same plugin has to save all the different layer types - might be nicer to have one plugin per layer type. Not sure how to think about that in terms of UI / API

Reply:
I'm actually not sure I see the need for this function at all in its current design (that is, a case in which the user is trying to save multiple layers, but we're also using the individual `napari_write_layer` hookspec _and_ specifying a certain plugin)

I imagined two cases for saving multiple layers: 

1) the case where we use `napari_get_writer` to give _single_ plugins a shot to save all selected layers on their own
2) the case where we dispatch individual `napari_write_layer` hooks from _multiple_  plugins to save each layer at a time... kinda like this function.  This is sort of a "try to save all of this stuff anyway you can".  but this is complicated and we need to do some more work to prepare for it (like creating a super-directory into which all plugins will be writing the output).

but in either case, I don't think we need to support the case where the user is trying to save multiple layers, using a single plugin, but _not_ using `napari_get_writer`.  let's leave that up to plugins.

Reply:
additional thought here:  we could implement our own `napari_get_writer` "fallback plugin" that accepts _any_ combination of `layer_type` strings, and dispatches each layer to the plugins via `napari_write_layer`, putting it all into a directory.